scripts/experiment_2/nbconverted/2_simulate_data_truncated.py

```diff
@@ -0,0 +1,291 @@
+
+
+ggplot(latent_data_UMAPencoded_df, aes(x='1',y='2'))     + geom_point(alpha=0.5)     + scale_color_brewer(type='qual', palette='Set2')     + ggtitle("Encoded input data")
+
+
+# ## Plot decoded input data using UMAP
+
+# In[9]:
+
+
+# Decode data back into gene space
+data_decoded = loaded_decode_model.predict_on_batch(data_encoded_df)
+data_decoded_df = pd.DataFrame(data_decoded, index=data_encoded_df.index)
+
+# Plot
+data_decoded_UMAPencoded = model.transform(data_decoded_df)
+data_decoded_UMAPencoded_df = pd.DataFrame(data=data_decoded_UMAPencoded,
+                                         index=data_decoded_df.index,
+                                         columns=['1','2'])
+
+
+ggplot(data_decoded_UMAPencoded_df, aes(x='1',y='2'))     + geom_point(alpha=0.5)     + scale_color_brewer(type='qual', palette='Set2')     + ggtitle("Decoded input data")
+
+
+# ## Simulate data
+# 
+# Generate new simulated data by sampling from the distribution of latent space features.  In other words, for each latent space feature get the mean and standard deviation.  Then we can generate a new sample by sampling from a distribution with this mean and standard deviation.
+
+# In[10]:
+
+
+# Simulate data
+
+# Encode into latent space
+data_encoded = loaded_model.predict_on_batch(normalized_data)
+data_encoded_df = pd.DataFrame(data_encoded, index=normalized_data.index)
+
+latent_dim = data_encoded_df.shape[1]
+
+# Get mean and standard deviation per encoded feature
+encoded_means = data_encoded_df.mean(axis=0)
+
+encoded_stds = data_encoded_df.std(axis=0)
+
+# Generate samples 
+new_data = np.zeros([num_simulated_samples,latent_dim])
+for j in range(latent_dim):
+    # Use mean and std for feature
+    new_data[:,j] = np.random.normal(encoded_means[j], encoded_stds[j], num_simulated_samples) 
+    
+    # Use standard normal
+    #new_data[:,j] = np.random.normal(0, 1, num_simulated_samples)
+    
+new_data_df = pd.DataFrame(data=new_data)
+
+# Decode N samples
+new_data_decoded = loaded_decode_model.predict_on_batch(new_data_df)
+simulated_data = pd.DataFrame(data=new_data_decoded)
+
+simulated_data.head(10)
+
+
+# In[11]:
+
+
+# Randomly select subset of genes 
+subset_simulated_data = simulated_data.sample(n=num_dims, axis=1)
+subset_simulated_data.head()
+
+
+# ## Plot simulated data using UMAP
+# 
+# Note: we will use the same UMAP mapping for the input and simulated data to ensure they are plotted on the same space.
+
+# In[12]:
+
+
+# UMAP embedding
+simulated_data_UMAPencoded = umap.UMAP(random_state=randomState).fit_transform(subset_simulated_data)
+simulated_data_UMAPencoded_df = pd.DataFrame(data=simulated_data_UMAPencoded,
+                                         index=simulated_data.index,
+                                         columns=['1','2'])
+
+
+ggplot(simulated_data_UMAPencoded_df, aes(x='1',y='2'))     + geom_point(alpha=0.5)     + scale_color_brewer(type='qual', palette='Set2')     + ggtitle("Simulated data")
```

Comment:
You're correct, the input and simulated data are not on the same set of UMAP axes because I subsampled the genes in the simulated data so input has 5549 genes and simulated has 5000 genes.  Moving forward it does not look like we will need to do this truncation, so I will be able to make a plot with the input and simulated data in one figure.  Good suggestion